### ISSUE TITLE
Make it easier to access a CSRF token in automated tests

### ISSUE THREAD

**Issue by singingwolfboy** [lepture/flask-wtf#227]
If you want to run your automated tests with CSRF enabled (which is a good idea if it's enabled in production), there's no good built-in way to do so. Even the tests for this project [use regular expressions to parse the CSRF token out of the page](https://github.com/lepture/flask-wtf/blob/3c9dcf5cc/tests/test_csrf.py#L10-L18), which is brittle and confusing. It would be better to provide some way to access the CSRF token in the Flask test client itself.

I've written a GitHub Gist that walks though how I implemented this myself, but maybe Flask-WTF could change some internals to make it cleaner and easier? https://gist.github.com/singingwolfboy/2fca1de64950d5dfed72

**Comment by davidism** [comment]
This will be fixed when #264 is merged and released. Among other things, it caches the signed token for each request in `request.csrf_token`. The raw token was already available in `session['csrf_token']`.

~~~python
with app.text_client() as c:
    rv = c.post('/')
    signed_token = request.csrf_token
    raw_token = session['csrf_token']
~~~

**Comment by davidism** [comment]
Moved the signed token to `g` since that's what `g` is there for.
